src/components/ActionSheetAwareScrollView/ActionSheetKeyboardSpace.js

```diff
@@ -0,0 +1,237 @@
+import React, {
+    useContext, useEffect, useMemo,
+} from 'react';
+import {Dimensions, Keyboard} from 'react-native';
+import Reanimated, {
+    useWorkletCallback,
+    Easing,
+    KeyboardState,
+    useAnimatedKeyboard,
+    useAnimatedStyle, useDerivedValue, withTiming,
+} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import styles from '../../styles/styles';
+import {Actions, States, ActionSheetAwareScrollViewContext} from './ActionSheetAwareScrollViewContext';
+
+const config = {
+    duration: 350,
+    easing: Easing.bezier(0.33, 0.01, 0, 1),
+};
+
+function ActionSheetKeyboardSpace(props) {
+    const safeArea = useSafeAreaInsets();
+    const keyboard = useAnimatedKeyboard();
+
+    // similar to using `global` in worklet but it's just a local object
+    const ctx = useMemo(() => ({}), []);
+    const windowHeight = Dimensions.get('screen').height;
+    const {
+        currentActionSheetState, transitionActionSheetStateWorklet: transition, transitionActionSheetState, resetStateMachine,
+    } = useContext(ActionSheetAwareScrollViewContext);
+
+    useEffect(() => {
+        // sometimes it might trigger multiple times the same actions
+        let lastAction = null;
+        const onDidShow = () => {
+            if (lastAction === 'keyboardDidShow') {
+                return;
+            }
+
+            lastAction = 'keyboardDidShow';
+
+            transitionActionSheetState({
+                type: Actions.ON_KEYBOARD_OPEN,
+            });
+        };
+        const removeDidShow = Keyboard.addListener('keyboardDidShow', onDidShow);
+
+        const onDidHide = () => {
+            if (lastAction === 'keyboardDidHide') {
+                return;
+            }
+
+            lastAction = 'keyboardDidHide';
+
+            transitionActionSheetState({
+                type: Actions.ON_KEYBOARD_CLOSE,
+            });
+        };
+        const removeDidHide = Keyboard.addListener('keyboardDidHide', onDidHide);
+
+        return () => {
+            resetStateMachine();
+
+            // We try to use the new API first, but it doesn't work all the time
```

Comment:
This seems like a workaround, can we try to fix the root cause instead?

Reply:
I'll look into that again, but it seems like it no longer happens for me, will play around it again